### PR TITLE
feat(provider): add native Wise multi-currency provider integration

### PR DESCRIPTION
**Pull request by jestinjoshi**
## Summary

- Adds a complete native Wise personal API token provider integration, following the same architecture as the Up Bank and Mercury providers
- `Provider::Wise` HTTP client with bearer auth, 365-day statement windowing (Wise caps at ~469 days), exponential-backoff retries, and typed error hierarchy
- `WiseItem` / `WiseAccount` models with `Encryptable`, `Syncable`, `Provided`, `Unlinking` concerns; scopes for linked/unlinked accounts
- `WiseItem::Importer` fetches profiles then balance jars then transaction statements; deduplicates by `referenceNumber`; prunes removed accounts
- `WiseItem::Syncer` staged sync (import then process then schedule) with pending-setup detection and `DebugLogEntry`-backed error capture throughout
- `WiseAccount::Processor` posts balances and transactions to `Account::ProviderImportAdapter` with correct sign convention (`amount.value` is signed in the Wise API; negated to match Sure convention where negative = inflow)
- `WiseItemsController` full CRUD plus select/link/setup account flows with admin guards and Turbo Stream responses
- `AccountsController` and `Settings::ProvidersController` wired for Wise panel and sync stats
- `Provider::WiseAdapter` registered; `wise` added to `DataEnrichment`, `ProviderMerchant`, `ProviderConnectionStatus`, and `Family::FinancialDataReset`
- Full locale file, settings panel view, and accounts index partial
- Fixtures and unit tests for processor sign convention and data helpers

## Notes

- Wise API tokens are long-lived and do not expire unless revoked by the user -- no refresh token rotation needed
- Sign convention: `amount.value` from the Wise statement API is already signed (positive = credit/inflow, negative = debit/outflow); Sure convention is inverted, so `signed_amount = -amount_value`
- Token setup path: Account Settings -> Connect and manage apps -> API tokens -> Create token with Read only access
- Pagination uses date-range windowing (365-day chunks) rather than cursor pagination -- safe from infinite-loop edge cases by design

## Test plan

- [ ] Connect a Wise account via Settings -> Providers -> Wise and verify sync completes
- [ ] Confirm balance and transactions appear on the Accounts page after linking
- [ ] Verify multi-currency balances (e.g. USD + EUR + GBP) each create separate linked accounts with correct currency
- [ ] Test `select_existing_account` flow from an existing account to link it to a Wise balance
- [ ] Verify `setup_accounts` flow creates accounts with correct types (Depository / Investment)
- [ ] Revoke the API token in Wise and confirm the item status flips to `requires_update`
- [ ] Run financial data reset and confirm Wise items and accounts are cleaned up

Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added Wise as a supported provider, including new provider settings panel, Wise provider panel, and Wise-aware status summaries.
  * Introduced a full Wise integration: create/edit/sync Wise connections plus account linking and guided setup dialogs.
  * Added Wise syncing for balances and transaction imports with per-item sync visibility.

* **Bug Fixes**
  * Updated provider summaries, accounts empty states, and reset/sync-health reporting to correctly include Wise when configured.

* **Tests**
  * Added coverage for the Wise client, account processing, and Wise data parsing.

### CODE REVIEW

File: app/controllers/accounts_controller.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@ def index
     @indexa_capital_items = visible_provider_items(family.indexa_capital_items.ordered.includes(:syncs, :indexa_capital_accounts))
     @sophtron_items = visible_provider_items(family.sophtron_items.ordered.includes(:syncs, :sophtron_accounts))
     @binance_items = visible_provider_items(family.binance_items.ordered.includes(:binance_accounts, :accounts, :syncs))
+    @wise_items = visible_provider_items(family.wise_items.ordered.includes(:syncs, :wise_accounts))
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply app/controllers/accounts_controller.rb]: Fixed in 323f15ce. Added `@wise_account_counts_map` built from preloaded data in `build_sync_stats_maps` (following the Brex pattern). Includes updated to `includes(:syncs, :accounts, wise_accounts: :account_provider)` so `account_provider.present?` checks in the map builder hit no DB. View reads from the map with a fallback to the instance methods for non-index contexts.

File: app/models/provider/wise_adapter.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  end
+
+  def self.connection_configs(family:)
+    return [] unless family.can_connect_wise?
+
+    [ {
+      key: "wise",
+      name: "Wise",
+      description: "Connect to your Wise multi-currency account",
+      can_connect: true,
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply app/models/provider/wise_adapter.rb:18]: Fixed in 323f15ce. `name` and `description` now use `I18n.t("wise_items.provider_config.name")` and `I18n.t("wise_items.provider_config.description")`. Keys added to `config/locales/views/wise_items/en.yml`. The hardcoded "Wise Connection" default in the controller and `WiseConnectable#create_wise_item!` is also now localized via `I18n.t("wise_items.default_name")`.

File: app/models/provider/wise_adapter.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      new_account_path: ->(accountable_type, return_to) {
+        Rails.application.routes.url_helpers.select_accounts_wise_items_path(
+          accountable_type: accountable_type,
+          return_to: return_to
+        )
+      },
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply app/models/provider/wise_adapter.rb:24]: Fixed in 323f15ce. Added `safe_return_to_path` (identical guard to Mercury/Lunchflow: rejects absolute URLs and non-`/` paths to prevent open redirect). `select_accounts` now forwards the sanitized value to `setup_accounts_wise_item_path`, and `complete_account_setup` redirects to `return_to || accounts_path`.

File: app/models/provider/wise.rb [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    def capture_response_error(reason, response)
+      DebugLogEntry.capture(
+        category: "provider_sync",
+        level: "error",
+        message: "Wise API #{reason} (#{response.code})",
+        source: self.class.name,
+        provider_key: "wise",
+        metadata: { status: response.code, body: response.body.to_s.first(1000) }
+      )
+    end
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply app/models/provider/wise.rb:167]: Fixed in 323f15ce. `capture_response_error` now parses the JSON response and extracts only the `error` and `message` fields (both typically non-sensitive Wise error codes/descriptions). Raw body is no longer logged. `DebugLogEntry` metadata is now `{ status: response.code, error: ..., message: ... }.compact`.

File: app/models/wise_item/importer.rb [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+            category: "provider_sync",
+            level: "error",
+            message: "WiseItem::Importer - Failed to import balance #{balance[:id]}: #{e.message}",
+            source: "WiseItem::Importer",
+            provider_key: "wise"
+          )
+          stats["accounts_skipped"] = stats.fetch("accounts_skipped", 0) + 1
+        end
+      end
+    end
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply app/models/wise_item/importer.rb:109]: Fixed in 323f15ce. Two changes: (1) `list_balances` is now wrapped in its own rescue — `AuthenticationError` re-raises, any other error logs via `DebugLogEntry` and returns early, leaving other profiles unaffected. (2) `safe_id` is captured before the inner `begin` block using a hash-safe accessor, so the rescue can never raise a secondary `NoMethodError`.

File: app/models/wise_item/importer.rb [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    rescue => e
+      register_error(type: "transaction_import_error", message: e.message, context: { wise_account_id: wise_account.id })
+      DebugLogEntry.capture(
+        category: "provider_sync",
+        level: "warn",
+        message: "WiseItem::Importer - Failed to fetch transactions for wise_account #{wise_account.id}: #{e.message}",
+        source: "WiseItem::Importer",
+        provider_key: "wise"
+      )
+    end
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply app/models/wise_item/importer.rb:165]: Fixed in 323f15ce. Added `rescue Provider::Wise::AuthenticationError; raise` before the generic rescue in `import_transactions`. Since `AuthenticationError < Error < StandardError`, the bare rescue was catching it and logging it as a transaction warning instead of propagating it up to `WiseItem::Syncer#perform_sync`, which is where `wise_item.update!(status: :requires_update)` lives.

File: app/models/wise_item/importer.rb [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+    # Wise's referenceNumber is unique per transaction. Fall back to a
+    # content hash for older entries that may lack it.
+    def transaction_key(txn)
+      txn = txn.with_indifferent_access if txn.is_a?(Hash)
+      ref = txn[:referenceNumber]
+      return ref if ref.present?
+
+      [ txn[:date], txn[:type], txn.dig(:amount, :value), txn.dig(:amount, :currency) ].join("-")
+    end
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply app/models/wise_item/importer.rb:194]: Fixed in 323f15ce. Added `details[:description].to_s` as a fifth segment to the fallback key in both `WiseItem::Importer#transaction_key` and `WiseAccount::Processor#transaction_fallback_key`. Same-day, same-type, same-amount transfers to different counterparties (the scenario described) will now produce distinct keys.

File: config/locales/views/settings/en.yml [hunk 8]
```diff
@@ -1,554 +1,555 @@
...
-        token_label: Token
-        token_placeholder_new: Enter your IBKR Flex Web Service Token
-        token_placeholder_existing: Leave blank to keep the current Token
-        save_configuration: Save Configuration
-        update_configuration: Update Configuration
-        status_configured_prefix: "%{summary}. Visit the"
-        accounts_tab: Accounts
-        status_configured_suffix: tab to manage discovered accounts.
-        not_configured: Not configured.
+---
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply config/locales/views/settings/en.yml]: Fixed in 323f15ce. Normalized to LF-only using a binary read/replace pass. File now has 0 CRLF sequences.

File: config/locales/views/wise_items/en.yml [hunk 9]
```diff
@@ -0,0 +1,142 @@
+---
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply config/locales/views/wise_items/en.yml]: Fixed in 323f15ce. Same treatment as settings/en.yml: normalized to LF-only. File now has 0 CRLF sequences.

File: config/locales/views/wise_items/en.yml [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+        one: "The selected account is already linked"
+        other: "All %{count} selected accounts are already linked"
+      partial_success: "Successfully linked %{created_count} account(s). %{already_linked_count} account(s) were already linked."
+      partial_invalid: "Successfully linked %{created_count} account(s), %{already_linked_count} were already linked, %{invalid_count} had invalid names."
+      no_accounts_selected: "Please select at least one account"
+      no_api_token: "Wise API token not found. Please configure it in Provider Settings."
+      link_failed: "Failed to link accounts"
+      success:
+        one: "Successfully linked %{count} account"
+        other: "Successfully linked %{count} accounts"
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply config/locales/views/wise_items/en.yml]: Fixed in 323f15ce. `link_accounts` now branches on `invalid_count` first: uses `partial_invalid` (which includes all three counts) when any accounts had invalid names, falls back to `partial_success` only when `already_linked_count > 0` and `invalid_count == 0`. The `partial_invalid` locale key was already present but was never reached.

File: test/models/provider/wise_test.rb [hunk 11]
```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class Provider::WiseTest < ActiveSupport::TestCase
+  test "raises ConfigurationError when api_token is blank" do
+    assert_raises(Provider::Wise::ConfigurationError) do
+      Provider::Wise.new(api_token: nil)
+    end
+  end
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply test/models/provider/wise_test.rb]: Fixed in 323f15ce. The original test was renamed to `raises ConfigurationError when api_token is nil`. Two new cases added: empty string (`""`) and whitespace-only (`"   "`). Both hit the `blank?` check in `validate_configuration!` and raise `ConfigurationError`.

**Comment by jestinjoshi** [comment]
## CodeRabbit fixes — commit 323f15ce

All 10 actionable comments addressed. Summary:

**Major:**
- **N+1 queries** — precomputed `@wise_account_counts_map` in `build_sync_stats_maps`; includes updated to `wise_accounts: :account_provider`; view reads from map
- **PII in logs** — `capture_response_error` now extracts `error`/`message` fields from parsed JSON; raw body no longer logged
- **Per-profile isolation** — `list_balances` failure logs and returns early instead of aborting all profiles; `safe_id` captured before `begin` block
- **AuthenticationError swallowed** — added `rescue Provider::Wise::AuthenticationError; raise` before generic rescue in `import_transactions`
- **Dedup key collision** — `details[:description]` added to fallback key in both importer and processor

**Minor:**
- **return_to not threaded** — `safe_return_to_path` added (same XSS guard as Mercury/Lunchflow); `select_accounts` forwards it; `complete_account_setup` uses it for redirect
- **Hardcoded strings** — `WiseAdapter` name/description use `I18n.t()`; "Wise Connection" default localized
- **Wrong locale key** — `link_accounts` now uses `partial_invalid` when `invalid_count > 0`
- **CRLF line endings** — normalized to LF in both locale files
- **Blank token tests** — added empty string and whitespace-only cases

**Also addressed from nitpick comments:**
- Extracted `update_sync_status` helper in syncer (6 repetitions)
- `unlinking.rb` uses `DebugLogEntry.capture` (was `Rails.logger.warn`)
- `processor.rb` DebugLogEntry includes `family:` and `metadata:`
- `linked/unlinked_accounts_count` delegates to existing scope methods
- Importer uses `wise_item.linked_wise_accounts` instead of re-joining
- `DebugLogEntry.capture` on retry exhaustion in `Provider::Wise`
- Unsupported-type tests for `parse_decimal`/`parse_date`

**Comment by jestinjoshi** [comment]
CodeRabbit fixes in 323f15ce - all 10 actionable comments addressed (N+1 queries, PII in logs, per-profile isolation, AuthenticationError propagation, dedup key, return_to threading, hardcoded strings, locale key, CRLF, blank token tests). Nitpick items also addressed: update_sync_status helper, DebugLogEntry in unlinking.rb, family/metadata in processor.rb, delegated count methods, linked_wise_accounts in importer, retry exhaustion logging, unsupported-type tests.

File: app/controllers/wise_items_controller.rb [hunk 12]
```diff
@@ -0,0 +1,326 @@
...
+
+    def safe_return_to_path
+      return nil if params[:return_to].blank?
+
+      return_to = params[:return_to].to_s
+
+      begin
+        uri = URI.parse(return_to)
+        return nil if uri.scheme.present?
+        return nil unless return_to.start_with?("/")
```
[the remark this answers is not in the document]
Comment (jestinjoshi) [reply app/controllers/wise_items_controller.rb:320]: Fixed in commit 29aa9aa9. Added \
eturn nil if uri.host.present?\ as a third guard: \URI.parse('//evil.com').scheme\ is nil and the path starts with '/', so the two prior checks were insufficient. The host check closes the protocol-relative URL bypass.

[the remark this answers is not in the document]
Comment (jestinjoshi) [reply app/controllers/wise_items_controller.rb:320]: Fixed in commit 29aa9aa9. Added 
eturn nil if uri.host.present? as a third guard: URI.parse('//evil.com').scheme is nil and the path starts with '/', so the two prior checks were insufficient. The host check closes the protocol-relative URL bypass.

**Comment by sure-design** [comment]
Reviewed: a large but well-organized integration that mirrors the existing Up Bank/Mercury provider architecture, adds no new gem dependencies, and has a fully green CI run. All prior CodeRabbit and security-bot findings (N+1 queries, PII in logs, a swallowed `AuthenticationError`, dedup key collisions, an open-redirect gap, CRLF line endings, etc.) have been resolved across the follow-up commits — good iteration. One gap: there's no Minitest coverage for `WiseItemsController` itself (CRUD, select/link/setup-account flows, admin guards) — the included tests only cover the provider client, processor sign-convention, and data-helper parsing. Given the size of the surface added to that controller, please add controller-level behavioral tests before merge.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jjmata** [comment]
Reviewed against the Up/Mercury patterns as requested.

**Sign convention** — checks out. `WiseAccount::Processor#process_transaction` does `signed_amount = -amount_value` (app/models/wise_account/processor.rb). That's the same negation `UpEntry::Processor#amount` and `MercuryEntry::Processor#amount` do to flip their provider's native "negative=out/positive=in" convention into Sure's "negative=inflow/positive=outflow." Consistent with existing providers.

**Encrypted credentials — one gap found.** `WiseItem` does `encrypts :api_token` (non-deterministic) inside the `if encryption_ready?` guard, which is fine mechanically, but every other single-token provider item in the codebase (`BrexItem#token`, `MercuryItem#token`, `PlaidItem#access_token`, `IndexaCapitalItem#api_token` — literally the same field name) uses `encrypts ..., deterministic: true`. I didn't find any code path that queries `WiseItem` by token value, so this isn't a functional bug, but it's worth confirming it's intentional rather than an oversight, since it's the one place this PR diverges from the established convention for that field.

More importantly: **`WiseItem#raw_payload`, `WiseAccount#raw_payload`, and `WiseAccount#raw_transactions_payload` (added in `db/migrate/20260702000001_create_wise_items_and_accounts.rb`) are never wrapped in `encrypts`.** Every comparable model in the repo encrypts these same-purpose columns — `MercuryAccount`, `UpAccount`, `PlaidAccount`, `BrexAccount`, `BinanceAccount`, `CoinbaseAccount`, `KrakenAccount`, `SimplefinAccount`, `AkahuAccount`, `EnableBankingAccount`, `IbkrAccount`, `SnaptradeAccount` all call `encrypts :raw_payload` / `encrypts :raw_transactions_payload`, and `AkahuItem`/`BrexItem`/`EnableBankingItem`/`LunchflowItem`/`PlaidItem`/`SimplefinItem` do the same for their item-level `raw_payload`. For Wise, these jsonb columns hold full statement/transaction payloads (descriptions, amounts, dates) and would be stored in plaintext at rest. This looks like an oversight rather than a deliberate choice and should be fixed to match the repo's at-rest encryption convention before merge.

**Error handling / DebugLogEntry** — solid, and actually goes a bit further than Mercury's client. `Provider::Wise` has a typed error hierarchy (`ConfigurationError`/`AuthenticationError`/`RetryableResponseError`), exponential-backoff retries with jitter, and `DebugLogEntry.capture` on both retry-exhaustion and unexpected-response paths (`provider_sync` category, `provider_key: "wise"`). `WiseItem::Syncer` correctly flips status to `requires_update` on `AuthenticationError`, matching the "revoke token" item in the test plan. `WiseAccount::Processor#process_transactions` also wraps each transaction in its own rescue + `DebugLogEntry.capture` so one bad record doesn't abort the whole batch — good pattern.

Net: architecture and sign handling look right and mirror the Up/Mercury precedent well; the missing `encrypts` on the raw payload columns is the one thing I'd block on.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jestinjoshi** [comment]
Synced findings from PR #2534 across:

- `turbo_prefetch: "false"` on provider links in `_method_selector.html.erb` and `select_provider.html.erb`

Everything else from the Questrade alignment was already in place from earlier commits.

@jjmata @sokiee @sure-admin

**Comment by sure-design** [comment]
Following up on @jjmata's 2026-07-04 review: the blocking encryption gap doesn't look addressed yet — `WiseItem#raw_payload`, `WiseAccount#raw_payload`, and `WiseAccount#raw_transactions_payload` (from `db/migrate/20260702000001_create_wise_items_and_accounts.rb`) still aren't wrapped in `encrypts`, unlike every comparable provider model (Mercury/Up/Plaid/Brex/etc.) that encrypts these same-purpose columns. Worth fixing before merge since these hold full statement/transaction payloads in plaintext at rest.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jestinjoshi** [comment]
@sure-design this was addressed in `1e269557`:

- `WiseItem`: `encrypts :api_token, deterministic: true` and `encrypts :raw_payload`
- `WiseAccount`: `include Encryptable` + `encrypts :raw_payload` and `encrypts :raw_transactions_payload`

Matches the Mercury/Up pattern exactly.

**Comment by sure-design** [comment]
Heads up: the `security-scan` check is currently **failing** on the head commit (run [28901191142/job/85738085404](https://github.com/we-promise/sure/actions/runs/28901191142/job/85738085404)), while the other CI jobs (test_unit, lint, etc.) are green. This appeared after the last review discussion here (which focused on the encryption gap, now fixed in `1e269557`) and hasn't been acknowledged yet. Given the similar "Credential in URL" pipelock finding that came up on the parallel Wise PR (#2433), it's worth checking whether this is the same false-positive pattern (needing a `# pipelock:ignore` with justification) or a real finding, before merge.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jjmata** [comment]
Can you review the merge conflicts @jestinjoshi? 🙏

**Comment by jestinjoshi** [comment]
Found PR #2433 from @xBlaz3kx implementing the same Wise integration. Features not yet covered in this PR:
- JAR/SAVINGS balances (v4 endpoint with `types=SAVINGS`, `totalWorth` for balance)
- INTERBALANCE activities routed to both sides and linked as Transfer records
- Income vs expense detection via `targetAccount == recipientId`
- BALANCE_CASHBACK / BALANCE_ASSET_FEE activity types

Plan: merge this as the base, then port those features in a follow-up with @xBlaz3kx.

@jjmata @sokie

**Comment by xBlaz3kx** [comment]
@jestinjoshi can't we close this as duplicate instead of having two PRs open for the same provider integration? If my PR includes more features, I wouldn't want to resolve merge conflicts.. 

Thoughts @jjmata?

**Comment by jestinjoshi** [comment]
@xBlaz3kx fair concern.

- #2433 covers more Wise features 
- #2591 covers more of the coding conventions (encryption, retry/backoff, typed errors, test coverage, FinancialDataReset integration)

If you'd rather keep #2433 as the base, you're welcome to copy the encryption, retry/backoff, and test coverage over from #2591 directly. That way #2591 can close without losing that work.

@jjmata what do you think?

**Comment by sure-design** [comment]
The duplicate-effort discussion with #2433 (@xBlaz3kx) below is still unresolved — xBlaz3kx asked on 2026-07-08 whether this PR should close in favor of #2433 (which covers more Wise features: JAR/SAVINGS balances, INTERBALANCE transfers, cashback/fee activity types) since merging this first would leave them resolving conflicts, and that question to @jjmata hasn't gotten a reply in 4 days. Separately, CI is now fully green and the earlier encryption-at-rest and security-scan concerns look resolved. Flagging mainly so the two-PR situation doesn't stall indefinitely — worth a maintainer call on which branch is the merge target before more work lands on either side.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jjmata** [comment]
Decided to take #2433 first, @jestinjoshi ... sorry!

**Comment by sure-design** [comment]
## DS Drift Patrol — 1 finding

### Rule 4: `DS::Disclosure` summary indentation (missing `w-full`)

- **`app/views/wise_items/_wise_item.html.erb:6`**
  ```erb
  <% disclosure.with_summary_content do %>
    <div class="flex items-center justify-between gap-2">
  ```
  This is a new `DS::Disclosure.new(variant: :card, open: true)` callsite. The top-level element inside `with_summary_content` is a flex row (`justify-between`) but is missing `w-full`. Since `<summary>` renders as `display: list-item`, a flex child without `w-full` shrink-wraps to its content width instead of filling the summary, so the `justify-between` spacing between the title block and the trailing action icons (refresh/menu) will collapse instead of spreading to the row's full width.

  Suggested fix: add `w-full` to the flex row, matching the established pattern in `app/views/up_items/_up_item.html.erb`:
  ```erb
  <div class="flex items-center justify-between gap-2 w-full">
  ```

  Note: this same gap already exists in several older `_*.html.erb` provider-item partials (e.g. `plaid_items/_plaid_item.html.erb`, `snaptrade_items/_snaptrade_item.html.erb`), so it isn't unique to this PR — but this PR introduces a brand new file with the same gap, so flagging it here rather than letting it propagate further.




---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by sure-design** [comment]
**Correction to the DS Drift Patrol comment above.**

The flagged finding (`app/views/wise_items/_wise_item.html.erb:6` — `DS::Disclosure(variant: :card)` summary flex row missing `w-full`) is a **false positive**. `app/components/DS/disclosure.html.erb` already auto-wraps `summary_content` in `<div class="w-full">` for every non-`:default` variant, so the caller's own top-level element doesn't need `w-full` — the component guarantees the containing block is full-width regardless of the class on the caller's div.

No action needed here. Sorry for the noise — the "Rule 4" check that generated this finding is stale (it predates a fix to the component) and is being retired.



---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jestinjoshi** [comment]
Duplicate of #2433
